### PR TITLE
add function to link switch to router

### PR DESCRIPTION
**Pull request by jxiaobin**

### CODE REVIEW

**Comment by jxiaobin** [comment]
@vtolstov @hzhou8 could you help review? thanks!

**Comment by JackyCZJ** [comment]
Maybe you can add some test for your change.

File: client.go [hunk 1]
```diff
@@ -39,6 +39,8 @@ type Client interface {
 
 	// Add logical port PORT on SWITCH
 	LSPAdd(ls string, lsp string) (*OvnCommand, error)
+	// Link logical port to router
+	LSPLinkToRouter(ls string, lsp, lrp string) (*OvnCommand, error)
```
Comment (hzhou8) [inline client.go]: It is creating a new LSP and link to an existed LRP, so the API is better to be named as LSPAddToLRP.

File: logical_switch_port.go [hunk 2]
```diff
@@ -73,6 +73,58 @@ func (odbi *ovndb) lspAddImp(lsw, lsp string) (*OvnCommand, error) {
...
+
+	mutateOp := libovsdb.Operation{
+		Op:        opMutate,
+		Table:     tableLogicalSwitch,
+		Mutations: []interface{}{mutation},
+		Where:     []interface{}{condition},
+	}
+
+	options := make(map[string]string)
+	options["router-port"] = lrp
```
Comment (hzhou8) [inline logical_switch_port.go]: Would it be better to validate if the LRP exists?

File: logical_switch_port.go [hunk 3]
```diff
@@ -73,6 +73,58 @@ func (odbi *ovndb) lspAddImp(lsw, lsp string) (*OvnCommand, error) {
...
+	optMap, _ := libovsdb.NewOvsMap(options)
+	// set options
+	setOptionsOp := libovsdb.Operation{
```
Comment (hzhou8) [inline logical_switch_port.go]: Can this be simpler? Can we avoid this setOptionsOp by directly: port["options"] = options, as part of the insertOp?

**Comment by jxiaobin** [comment]
closing this pr, will submit another one to link switch to router in a transaction

**Pull request closed without merging**